docs/guide/cookbook/module.md

```diff
@@ -419,7 +419,58 @@ Routes to be added must be _Array_ type even if it only has one element as you c
 
 ![route_liked_borderline](../images/route_liked.png)
 
-### 2-4. Recipe D (Use hooks)
+### 2-4. Recipe D (Create hooks)
+As you know we have two main kinds of hooks `listener` and `mutator`. In this section, we will look into how to use these hooks to create our own functionality.
+
+1. Create the `hooks.ts` inside `./src/modules/example-module`.
+
+2. Import `hooks` from the `@vue-storefront/core/lib/hooks`:
+
+```js
+import { createListenerHook, createMutatorHook } from '@vue-storefront/core/lib/hooks'
+```
+
+3. Prepare our own `hooks`:
+```ts
+// abridged ...
+
+const {
+  hook: beforeExampleHook,
+  executor: beforeExampleExecutor
+} = createMutatorHook()
+
+// Also we can add some generic types to hooks then the first object is responsible for incoming data structure and the second shows us how to return data will be looking.
```

Comment:
`Also we can add some generic types to the hooks. The first argument is responsible for incoming data structure and the second one is type of returned data`